wally-registry-backend/src/auth.rs

```diff
@@ -47,6 +66,34 @@ fn match_api_key<T>(request: &Request<'_>, key: &str, result: T) -> Outcome<T, a
     }
 }
 
+async fn verify_github_token(request: &Request<'_>) -> Outcome<WriteAccess, anyhow::Error> {
+    let token: String = match request.headers().get_one("authorization") {
+        Some(key) if key.starts_with("Bearer ") => (key[6..].trim()).to_owned(),
+        _ => {
+            return Outcome::Failure((Status::Unauthorized, format_err!("Github auth required")));
+        }
+    };
+
+    let client = reqwest::Client::new();
+    let response = client
+        .get("https://api.github.com/user")
+        .header("accept", "application/json")
+        .header("user-agent", "wally")
```

Comment:
Should this be `UpliftGames/wally`? I vaguely remember GitHub requesting that you do that.

Reply:
If I understand the [documentation](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#user-agent-required) correctly then this is our oauth application name, not our repo name,.